File: samples/CS/06-ComboBox/RibbonItems.cs
using System;
using System.Windows.Forms;

namespace WinForms.Ribbon
{
    partial class RibbonItems
    {

        public void Init()
        {
            ButtonDropA.Click += new EventHandler<EventArgs>(_buttonDropA_ExecuteEvent);
            ButtonDropB.Click += new EventHandler<EventArgs>(_buttonDropB_ExecuteEvent);
            ButtonDropC.Click += new EventHandler<EventArgs>(_buttonDropC_ExecuteEvent);
            ButtonDropD.Click += new EventHandler<EventArgs>(_buttonDropD_ExecuteEvent);
            ButtonDropE.Click += new EventHandler<EventArgs>(_buttonDropE_ExecuteEvent);
            ButtonDropF.Click += new EventHandler<EventArgs>(_buttonDropF_ExecuteEvent);

            InitComboBoxes();
        }

        void _buttonDropA_ExecuteEvent(object sender, EventArgs e)
        {
            // get selected item index from combo box 1
            int selectedItemIndex = ComboBox1.SelectedItem;

            if (selectedItemIndex == -1)
            {
                MessageBox.Show("No item is selected in simple combo");
            }
            else
            {
                string label = ComboBox1.GalleryItemItemsSource[(int)selectedItemIndex].Label;
                MessageBox.Show("Selected item in simple combo is: " + label);
            }
        }

        void _buttonDropB_ExecuteEvent(object sender, EventArgs e)
        {
            // get string value from combo box 2
            string stringValue = ComboBox2.StringValue;
            MessageBox.Show("String value in advanced combo is: " + stringValue);
        }

        void _buttonDropC_ExecuteEvent(object sender, EventArgs e)
        {
            foreach (GalleryItemPropertySet propSet in ComboBox1.GalleryItemItemsSource)
            {
                string label = propSet.Label;
                MessageBox.Show("Label = " + label);
            }
        }

        void _buttonDropD_ExecuteEvent(object sender, EventArgs e)
        {
            ComboBox1.GalleryItemItemsSource.ChangedEvent += new EventHandler<CollectionChangedEventArgs>(_uiCollectionChangedEvent_ChangedEvent);
        }

        void _buttonDropE_ExecuteEvent(object sender, EventArgs e)
        {
            UICollection<GalleryItemPropertySet> itemsSource1 = ComboBox1.GalleryItemItemsSource;
            int count = itemsSource1.Count;
            count++;
            itemsSource1.Add(new GalleryItemPropertySet() { Label = "Label " + count.ToString(), CategoryId = -1 });
        }

        void _buttonDropF_ExecuteEvent(object sender, EventArgs e)
        {
            ComboBox1.GalleryItemItemsSource.ChangedEvent -= new EventHandler<CollectionChangedEventArgs>(_uiCollectionChangedEvent_ChangedEvent);
        }

        void _uiCollectionChangedEvent_ChangedEvent(object sender, CollectionChangedEventArgs e)
        {
            MessageBox.Show("Got ChangedEvent. Action = " + e.Action.ToString());
        }

        private void InitComboBoxes()
        {
            ComboBox1.RepresentativeString = "Label 1";
            ComboBox2.RepresentativeString = "XXXXXXXXXXX";
            ComboBox3.RepresentativeString = "XXXXXXXXXXX";

            ComboBox1.Label = "Simple Combo";
            ComboBox2.Label = "Advanced Combo";
            ComboBox3.Label = "Another Combo";

            ComboBox1.ItemsSourceReady += new EventHandler<EventArgs>(_comboBox1_ItemsSourceReady);

            ComboBox2.CategoriesReady += new EventHandler<EventArgs>(_comboBox2_CategoriesReady);
            ComboBox2.ItemsSourceReady += new EventHandler<EventArgs>(_comboBox2_ItemsSourceReady);

            ComboBox3.ItemsSourceReady += new EventHandler<EventArgs>(_comboBox3_ItemsSourceReady);
        }

        void _comboBox1_ItemsSourceReady(object sender, EventArgs e)
        {
            // set combobox1 items
            UICollection<GalleryItemPropertySet> itemsSource1 = ComboBox1.GalleryItemItemsSource;
            itemsSource1.Clear();
            itemsSource1.Add(new GalleryItemPropertySet() { Label = "Label 1", CategoryId = -1 });
            itemsSource1.Add(new GalleryItemPropertySet() { Label = "Label 2", CategoryId = -1 });
            itemsSource1.Add(new GalleryItemPropertySet() { Label = "Label 3", CategoryId = -1 });

        }

        void _comboBox2_CategoriesReady(object sender, EventArgs e)
        {
            // set _comboBox2 categories
            UICollection<GalleryItemPropertySet> categories2 = ComboBox2.GalleryCategories;
            categories2.Clear();
            categories2.Add(new GalleryItemPropertySet() { Label = "Category 1", CategoryId = 1 });
            categories2.Add(new GalleryItemPropertySet() { Label = "Category 2", CategoryId = 2 });
        }

        void _comboBox2_ItemsSourceReady(object sender, EventArgs e)
        {
            // set _comboBox2 items
            UICollection<GalleryItemPropertySet> itemsSource2 = ComboBox2.GalleryItemItemsSource;
            itemsSource2.Clear();
            itemsSource2.Add(new GalleryItemPropertySet() { Label = "Label 1", CategoryId = 1 });
            itemsSource2.Add(new GalleryItemPropertySet() { Label = "Label 2", CategoryId = 1 });
            itemsSource2.Add(new GalleryItemPropertySet() { Label = "Label 3", CategoryId = 2 });
        }

        void _comboBox3_ItemsSourceReady(object sender, EventArgs e)
        {
            // set combobox3 items
            UICollection<GalleryItemPropertySet> itemsSource3 = ComboBox3.GalleryItemItemsSource;
            itemsSource3.Clear();
            itemsSource3.Add(new GalleryItemPropertySet() { Label = "Label 1", CategoryId = -1 });
            itemsSource3.Add(new GalleryItemPropertySet() { Label = "Label 2", CategoryId = -1 });
            itemsSource3.Add(new GalleryItemPropertySet() { Label = "Label 3", CategoryId = -1 });
        }

        public void Load()
        {
        }

    }
}

File: samples/VB/06-ComboBox/RibbonItems.vb
Imports System

Namespace WinForms.Ribbon

    Partial Class RibbonItems

		Public Sub Init()

			AddHandler _ButtonDropA.Click, AddressOf _buttonDropA_ExecuteEvent
			AddHandler _ButtonDropB.Click, AddressOf _buttonDropB_ExecuteEvent
			AddHandler _ButtonDropC.Click, AddressOf _buttonDropC_ExecuteEvent
			AddHandler _ButtonDropD.Click, AddressOf _buttonDropD_ExecuteEvent
			AddHandler _ButtonDropE.Click, AddressOf _buttonDropE_ExecuteEvent
			AddHandler _ButtonDropF.Click, AddressOf _buttonDropF_ExecuteEvent

			InitComboBoxes()
		End Sub

		Private Sub _buttonDropA_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			' get selected item index from combo box 1
			Dim selectedItemIndex As Integer = _ComboBox1.SelectedItem

			If selectedItemIndex = -1 Then
				MessageBox.Show("No item is selected in simple combo")
			Else
				Dim label As String
				label = _ComboBox1.GalleryItemItemsSource.Item(selectedItemIndex).Label
				MessageBox.Show("Selected item in simple combo is: " & label)
			End If
		End Sub

		Private Sub _buttonDropB_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			' get string value from combo box 2
			Dim stringValue As String = _ComboBox2.StringValue
			MessageBox.Show("String value in advanced combo is: " & stringValue)
		End Sub

		Private Sub _buttonDropC_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			' enumerate over items
			For Each propSet As GalleryItemPropertySet In ComboBox1.GalleryItemItemsSource
				Dim label As String = propSet.Label
				MessageBox.Show("Label = " & label)
			Next
		End Sub

		Private Sub _buttonDropD_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			AddHandler _ComboBox1.GalleryItemItemsSource.ChangedEvent, AddressOf _uiCollectionChangedEvent_ChangedEvent
		End Sub

		Private Sub _buttonDropE_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			Dim itemsSource1 As UICollection(Of GalleryItemPropertySet) = _ComboBox1.GalleryItemItemsSource
			Dim count As Integer
			count = itemsSource1.Count
			count += 1
			itemsSource1.Add(New GalleryItemPropertySet() With {.Label = "Label " & count.ToString(), .CategoryId = -1})
		End Sub

		Private Sub _buttonDropF_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			RemoveHandler _ComboBox1.GalleryItemItemsSource.ChangedEvent, AddressOf _uiCollectionChangedEvent_ChangedEvent
		End Sub

		Private Sub _uiCollectionChangedEvent_ChangedEvent(ByVal sender As Object, ByVal e As CollectionChangedEventArgs)
			MessageBox.Show("Got ChangedEvent. Action = " & e.Action.ToString())
		End Sub

		Private Sub InitComboBoxes()
			_ComboBox1.RepresentativeString = "Label 1"
			_ComboBox2.RepresentativeString = "XXXXXXXXXXX"
			_ComboBox3.RepresentativeString = "XXXXXXXXXXX"

			_ComboBox1.Label = "Simple Combo"
			_ComboBox2.Label = "Advanced Combo"
			_ComboBox3.Label = "Another Combo"

			AddHandler _ComboBox1.ItemsSourceReady, AddressOf _comboBox1_ItemsSourceReady

			AddHandler _ComboBox2.ItemsSourceReady, AddressOf _comboBox2_CategoriesReady
			AddHandler _ComboBox2.ItemsSourceReady, AddressOf _comboBox2_ItemsSourceReady

			AddHandler _ComboBox3.ItemsSourceReady, AddressOf _comboBox3_ItemsSourceReady
		End Sub

		Private Sub _comboBox1_ItemsSourceReady()
			' set combobox1 items
			Dim itemsSource1 As UICollection(Of GalleryItemPropertySet) = _ComboBox1.GalleryItemItemsSource
			itemsSource1.Clear()
			itemsSource1.Add(New GalleryItemPropertySet() With {.Label = "Label 1", .CategoryId = -1})
			itemsSource1.Add(New GalleryItemPropertySet() With {.Label = "Label 2", .CategoryId = -1})
			itemsSource1.Add(New GalleryItemPropertySet() With {.Label = "Label 3", .CategoryId = -1})
		End Sub

		Private Sub _comboBox2_CategoriesReady()
			' set _comboBox2 categories
			Dim categories2 As UICollection(Of CategoriesPropertySet) = _ComboBox2.GalleryCategories
			categories2.Clear()
			categories2.Add(New CategoriesPropertySet() With {.Label = "Category 1", .CategoryId = 1})
			categories2.Add(New CategoriesPropertySet() With {.Label = "Category 2", .CategoryId = 2})
		End Sub

		Private Sub _comboBox2_ItemsSourceReady()
			' set _comboBox2 items
			Dim itemsSource2 As UICollection(Of GalleryItemPropertySet) = _ComboBox2.GalleryItemItemsSource
			itemsSource2.Clear()
			itemsSource2.Add(New GalleryItemPropertySet() With {.Label = "Label 1", .CategoryId = 1})
			itemsSource2.Add(New GalleryItemPropertySet() With {.Label = "Label 2", .CategoryId = 1})
			itemsSource2.Add(New GalleryItemPropertySet() With {.Label = "Label 3", .CategoryId = 2})
		End Sub

		Private Sub _comboBox3_ItemsSourceReady()
			' set combobox3 items
			Dim itemsSource3 As UICollection(Of GalleryItemPropertySet) = _ComboBox3.GalleryItemItemsSource
			itemsSource3.Clear()
			itemsSource3.Add(New GalleryItemPropertySet() With {.Label = "Label 1", .CategoryId = -1})
			itemsSource3.Add(New GalleryItemPropertySet() With {.Label = "Label 2", .CategoryId = -1})
			itemsSource3.Add(New GalleryItemPropertySet() With {.Label = "Label 3", .CategoryId = -1})
		End Sub

	End Class
End Namespace
